scripts-available/CDB_PG_Federated_Tables.sql

```diff
@@ -0,0 +1,295 @@
+----------------------------------------------------------------------
+-- Federated Tables management functions
+----------------------------------------------------------------------
+
+-- Take a config jsonb and transform it to an input suitable for
+-- _CDB_SetUp_User_PG_FDW_Server
+CREATE OR REPLACE FUNCTION @extschema@.__ft_credentials_to_user_mapping(input_config jsonb)
+RETURNS jsonb
+AS $$
+DECLARE
+    user_mapping jsonb;
+BEGIN
+    user_mapping := json_build_object('user_mapping',
+        jsonb_build_object(
+            'user', input_config->'credentials'->'username',
+            'password', input_config->'credentials'->'password'
+        )
+    );
+    RETURN (input_config - 'credentials')::jsonb || user_mapping;
+END
+$$
+LANGUAGE PLPGSQL IMMUTABLE PARALLEL SAFE;
+
+
+-- Take a config jsonb as input and return it augmented with default
+-- options
+CREATE OR REPLACE FUNCTION @extschema@.__ft_add_default_options(input_config jsonb)
+RETURNS jsonb
+AS $$
+DECLARE
+    default_options jsonb := '{
+        "extensions": "postgis",
+        "updatable": "false",
+        "use_remote_estimate": "true",
+        "fetch_size": "1000"
+    }';
+    server_config jsonb;
+BEGIN
+    server_config := default_options || to_jsonb(input_config->'server');
+    RETURN jsonb_set(input_config, '{server}'::text[], server_config);
+END
+$$
+LANGUAGE PLPGSQL IMMUTABLE PARALLEL SAFE;
+
+
+CREATE OR REPLACE FUNCTION @extschema@.__ft_assert_numeric(input_table regclass, colname name)
+RETURNS VOID
+AS $$
+BEGIN
```

Comment:
Do we need to check for NULLs too?

Reply:
input params? why? I don't think so but may be wrong

Reply:
I mean that  the column contents might be NULL.

Reply:
Yes, and there may be duplicates, but there's little we can do about it without an "aggressive" check to inform the user.

I'd add that check only if we see it does save us some "support time".